frontend/src/components/project/project.jsx

```diff
@@ -1,32 +1,86 @@
+import _ from 'lodash';
 import React from 'react';
+import Reflux from 'reflux';
+import {ProjectActions} from '../../actions/actions';
+import ProjectStore from '../../stores/projectStore';
+
+import {DragDropContext} from 'react-dnd';
+import HTML5Backend from 'react-dnd/modules/backends/HTML5';
+
+import CurrentSprintInfo from './currentSprintInfo.jsx';
+import NextSprintInfo from './nextSprintInfo.jsx';
+import Droparea from './droparea.jsx';
 import Backlog from './backlog.jsx';
-import CreateSprint from './createSprint.jsx';
-import {mockProjects} from '../../../tests/utils/fake.js';
 
 let Project = React.createClass({
+  mixins: [
+    Reflux.ListenerMixin
+  ],
 
   getInitialState() {
     return {
-      project: mockProjects(1, 2, 4)[0]
+      project: {},
+      currentSprint: {},
+      nextSprint: {},
+      tasksInCurrentSprint : [],
+      tasksForNextSprint: [],
+      tasksInBacklog: []
     };
   },
 
   componentDidMount() {
+    this.listenTo(ProjectStore, this.onStoreUpdate);
     let id = this.props.params.id;
-    // conduct fetch for project details using this projectId
+    ProjectActions.fetchProject(id);
+  },
+
+  onStoreUpdate(project) {
+    // we may also want to keep track of 'done tasks'; not doing it at the moment
+    let currentSprint = _.findWhere(project.sprints, {status: 'In Progress'});
+    let nextSprint = _.findWhere(project.sprints, {status: 'Not Started'});
+    let tasksInCurrentSprint;
+    if (currentSprint) {
+      tasksInCurrentSprint = _.where(project.tasks, {sprintId: currentSprint.id});
+    }
+    let tasksForNextSprint = _.where(project.tasks, {sprintId: nextSprint.id});
+    let tasksInBacklog = _.where(project.tasks, {sprintId: null});
+    this.setState({
+      project: project,
+      currentSprint: currentSprint,
+      nextSprint: nextSprint,
+      tasksInCurrentSprint : tasksInCurrentSprint,
+      tasksForNextSprint: tasksForNextSprint,
+      tasksInBacklog: tasksInBacklog
+    });
   },
 
   render() {
-    let tasks = this.state.project.tasks;
-    return (
-      <div className='project-view'>
-        <CreateSprint />
-        <hr />
-        <Backlog tasks={tasks} />
-      </div>
-    );
+    if (this.state.currentSprint) {
+      return (
+        <div className='project-view'>
+          <CurrentSprintInfo
+            sprint={this.state.currentSprint}
+            tasks={this.state.tasksInCurrentSprint}
+          />
+          <Droparea tasks={this.state.tasksForNextSprint} />
+          <hr />
+          <Backlog tasks={this.state.tasksInBacklog} />
+        </div>
+      );
+    } else {
+      return (
+        <div className='project-view'>
+          <NextSprintInfo
+            sprint={this.state.nextSprint}
+            tasks={this.state.tasksForNextSprint}
+          />
+          <Droparea tasks={this.state.tasksForNextSprint} />
+          <hr />
+          <Backlog tasks={this.state.tasksInBacklog} />
+        </div>
+      );
+    }
```

Comment:
I wonder whether there is a more concise way to write these if-else statements in JSX. This way looks repetitive.